src/history/hs_rec.c

```diff
@@ -356,7 +356,12 @@ __wt_hs_insert_updates(
         enable_reverse_modify = true;
 
         __wt_update_vector_clear(&updates);
-        WT_ASSERT(session, out_of_order_ts_updates.size == 0);
+        /*
+         * In the case that the onpage value is an out of order timestamp update, it remains in the
+         * stack. Clean it up.
+         */
+        WT_ASSERT(session, out_of_order_ts_updates.size == 0 || out_of_order_ts_updates.size == 1);
```

Comment:
Are we 1000% sure the only case in which `out_of_order_ts_updates.size == 1` is when the on page value is the same as the value on the stack? We could miss a potential issue if we are assuming this is the case. Can we verify this in the assert?

Reply:
Yes, we can. I can add it.